src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1665,15 +1665,19 @@ private BoundBlock BindBlockParts(BlockSyntax node, DiagnosticBag diagnostics)
         {
             var syntaxStatements = node.Statements;
             int nStatements = syntaxStatements.Count;
-
+            bool sawGotos = false;
             ArrayBuilder<BoundStatement> boundStatements = ArrayBuilder<BoundStatement>.GetInstance(nStatements);
 
             for (int i = 0; i < nStatements; i++)
             {
                 var boundStatement = BindStatement(syntaxStatements[i], diagnostics);
                 boundStatements.Add(boundStatement);
+
+                sawGotos |= boundStatement.Kind == BoundKind.GotoStatement;
```

Comment:
>boundStatement.Kind [](start = 28, length = 19)

Are we handling nested `goto`s?
```c#
{
    goto L1;
}
using expr;
L1: ;
```

Reply:
We weren't. I've re-written this as part of control flow as it makes the logic much simpler